src/redis_lib/mod.rs

```diff
@@ -10,44 +11,134 @@ use tracing::error;
 ///
 /// * REDIS_PORT: the port of redis, defaults to 6379
 /// * REDIS_HOST: the host of redis, ip or hostname or domain, defaults to 127.0.0.1
-pub fn get_connection() -> Option<Client> {
-    let redis_port: u16;
-    match env::var("REDIS_PORT") {
-        Ok(val) => match val.parse() {
-            Ok(value) => redis_port = value,
-            Err(err) => {
-                error!(error=err.to_string(), "Failed to parse int");
-                return None;
-            }
-        },
-        Err(_) => redis_port = 6379
-    };
-    let redis_hostname = match env::var("REDIS_HOST") {
-        Ok(val) => val,
-        Err(_) => "127.0.0.1".to_string()
-    };
-    let client;
-    match Client::open(format!("redis://{}:{}", redis_hostname, redis_port)) {
-        Ok(val) => client = val,
+pub async fn get_connection() -> Option<(Client, Connection)> {
+    let redis_port: Result<u16, _> = env::var("REDIS_PORT").unwrap_or("6379".to_owned()).parse();
+    if let Err(err) = redis_port {
+        error!(error = err.to_string(), "Failed to parse int");
+        return None;
+    }
+    let redis_port = redis_port.unwrap();
+
+    let redis_hostname = env::var("REDIS_HOST").unwrap_or("127.0.0.1".to_owned());
+
+    let client = Client::open(format!("redis://{redis_hostname}:{redis_port}"))
+        .map_err(|e| {
+            error!(error = e.to_string(), "Problem creating redis client");
+        })
+        .ok()?;
+
+    let connection = match client.get_async_connection().await {
         Err(err) => {
-            error!(error=err.to_string(), "Problem creating redis client");
+            error!(error = err.to_string(), "Failed connecting to Redis");
             return None;
         }
+        Ok(con) => con,
+    };
+
+    Some((client, connection))
```

Comment:
What? It's completely obvious when the service isn't alive, because the request will fail - opening a new connection for every single operation is completely unnecessary, and depending on how it's handled internally, might be a considerable slowdown.

Reply:
While this may be true in most cases, in our case the Bot is idle for 99% of it's lifetime and the connection to these services (redis and MySQL) is only done on very rare occasions. Keeping the connection open also uses ressources. The performance hit of opening one when needed is negligible compared to the ressources needed on idle when considering the time spend in idle.